source/dub/packagemanager.d

```diff
@@ -167,6 +167,25 @@ class PackageManager {
 		this.refresh();
 	}
 
+	/** Override (read-only) package cache paths to search for packages.
+
+		Cache paths have the same structure as the default cache paths, such as
+		".dub/packages/".
+
+		Note that previously set custom paths will be removed when setting this
+		property.
+	*/
+	@property void overrideCachePaths(NativePath[] override_cache_paths)
```

Comment:
Alternatively, we could add an enum flag to `customCachePaths that describes how the lookup path is customized; typically as appended, prepended or replaced.